src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -155,13 +155,28 @@ def variant_compatible_siblings(filenames, variant=None) -> Union[List[os.PathLi
     transformers_index_format = r"\d{5}-of-\d{5}"
 
     if variant is not None:
-        # `diffusion_pytorch_model.fp16.bin` as well as `model.fp16-00001-of-00002.safetensors`
+        # Examples:
+        # For transformers, `pytorch_model.fp16.bin` as well as `pytorch_model.fp16-00001-of-00004.bin`.
+        # For diffusers `diffusion_pytorch_model.fp16.bin` as well as `diffusion_pytorch_model-00001-of-00002.fp16.safetensors`
+        # These differences exist because `diffusers` delegates the process of loading sharded checkpoints
```

Comment:
It is true that diffusers use `accelerate` to load shared checkpoints. However, `accelerate` does decide the checkpoint name pattern, no?   why diffusers' checkpoint has a different pattern from the transformers' one?

Reply:
As far as my memory serves, I made it this way because `load_checkpoint_and_dispatch()` was unable to automatically discover the `index.json.` when it contained a variant, following the `transformers` format. This was happening because we were passing the folder containing the sharded checkpoints. 

In that case, the concerned folder would contain two index files, for [example](https://huggingface.co/fal/AuraFlow/tree/main/transformer) -- ".index.json", ".fp16.index.json". So, this needed small changes to `_add_variant()`. 

But I didn't realize that `load_checkpoint_and_dispatch()` can take an index_file path directly, too. 

In retrospect, I do find our checkpoint naming convention to be a tadbit better (it is only different when dealing with sharded checkpoints):

<img width="828" alt="image" src="https://github.com/user-attachments/assets/ee50b7cd-efce-4c93-8060-eed6bbcac944">

I would expect the index file to end with `index.json` and not with `index.fp16.json`. 

We can either change it now and add code to handle the previous pattern accordingly or keep things as is. This will, of course, go through rigorous test cases. LMK how you want this to be dealt with.

Reply:
just want to be clear, this difference (for both the checkpoint name and index name) is only for shared checkpoint + variant?

Reply:
Yes that is right. It is only different when we have variant in the sharded checkpoints.

Reply:
I think we can change our naming to match transformers then, and then deprecate the current format(since I think sharding + variant hasn't been used that much, no? auraflow is the only one that I'm aware and it is not even functioning on pipeline level yet